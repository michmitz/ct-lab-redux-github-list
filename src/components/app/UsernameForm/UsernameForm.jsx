import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setUsername, fetchUser, fetchFollowers, fetchRepos } from '../../../actions/githubActions';

const UsernameForm = () => {
  const dispatch = useDispatch();
  const userName = useSelector(state => state.username);
  
  const handleChange = ({ target }) => {
    dispatch(setUsername(target.value));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    
    dispatch(fetchUser(userName));
    dispatch(fetchFollowers(userName));
    dispatch(fetchRepos(userName));
  };

  return (
    <form onSubmit={handleSubmit}>
    <label>
    <textarea onChange={handleChange} placeholder="username"></textarea>
    </label>
    <button>Submit</button>
    </form>
  );
};

export default UsernameForm;
