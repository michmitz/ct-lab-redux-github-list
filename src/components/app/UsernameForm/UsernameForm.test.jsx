import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UsernameForm from './UsernameForm';
import { Provider } from 'react-redux';
import store from '../../../store';

describe('UsernameForm component', () => {
  afterEach(() => cleanup());
  it('renders UsernameForm', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <UsernameForm />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});