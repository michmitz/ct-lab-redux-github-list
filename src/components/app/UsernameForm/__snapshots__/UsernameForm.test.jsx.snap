// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernameForm component renders UsernameForm 1`] = `
<DocumentFragment>
  <form>
    <label>
      <textarea
        placeholder="username"
      />
    </label>
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
